File: api/gemini.ts
import { GoogleGenerativeAI } from "@google/generative-ai";

type Req = {
  body?: { prompt?: string };
};

export default async function handler(req: Req, res: any) {
  try {
    if (req?.body == null) {
      return res.status(400).json({ error: "Missing request body" });
    }

    const apiKey = process.env.GEMINI_API_KEY;
    if (!apiKey) {
      return res.status(500).json({ error: "Missing GEMINI_API_KEY" });
    }

    const { prompt } = req.body;
    if (!prompt) {
      return res.status(400).json({ error: "Missing prompt" });
    }

    const genAI = new GoogleGenerativeAI(apiKey);
    
    // Architect Update: March 2026 Model Standard
    const model = genAI.getGenerativeModel({
      model: "gemini-3.1-pro-preview", 
      generationConfig: {
        responseMimeType: "application/json",
      },
    });

    const result = await model.generateContent(prompt);
    let raw = result.response.text() || "";

    // Design Check: Remove common LLM markdown wrapping if present
    raw = raw.replace(/```json|```/g, "").trim();

    // Guarantee we always return JSON
    let parsed: any;
    try {
      parsed = JSON.parse(raw);
    } catch (parseErr) {
      console.error("Parse Error Raw Output:", raw);
      return res.status(500).json({
        error: "Gemini returned non-JSON architecture",
        raw,
      });
    }

    return res.status(200).json(parsed);
  } catch (err: any) {
    console.error("Gemini API error:", err);
    return res.status(500).json({ error: err?.message || "VSE Clarity Core failure" });
  }
}

File: api/gemini.py
import os
import json
from google import genai

def handler(request):
    api_key = os.getenv("GEMINI_API_KEY") or os.getenv("GOOGLE_API_KEY")

    if not api_key:
        return {
            "statusCode": 500,
            "headers": {"Content-Type": "application/json"},
            "body": json.dumps({"error": "Missing GEMINI_API_KEY"})
        }

    client = genai.Client(api_key=api_key)

    # Parse JSON body from request to get the prompt
    try:
        body = json.loads(request.body or "{}")
        prompt = body.get("prompt", "")
    except Exception:
        prompt = ""

    if not prompt:
        return {
            "statusCode": 400,
            "headers": {"Content-Type": "application/json"},
            "body": json.dumps({"error": "Missing prompt"})
        }

    # Call Gemini with the full orchestration prompt
    response = client.models.generate_content(
        model="gemini-2.5-pro",
        contents=prompt,
    )

    # response.text is already the JSON object you saw in DevTools.
    # Wrap it in { "text": ... } so the TS client continues to work unchanged.
    orchestration_json = response.text  # string

    return {
        "statusCode": 200,
        "headers": {"Content-Type": "application/json"},
        "body": json.dumps({"text": orchestration_json})
    }